### PR TITLE
use label marker for distance and closeness features

### PR DESCRIPTION
**Pull request by arnej27959**
* distance feature now has a two arguments version, allowing
  the user to explicitly specify label or field
* extend closeness feature the same ways as distance feature
* add unit test for NNS closeness

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

as discussed earlier today.
@geirst please review
@havardpe FYI
system test must be updated when merging this, i have the changes ready and will make PR.

### CODE REVIEW

File: searchlib/src/tests/features/nns_closeness/nns_closeness_test.cpp [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+};
+
+struct FeatureDumpFixture : public IDumpFeatureVisitor {
+    virtual void visitDumpFeature(const vespalib::string &) override {
+        TEST_ERROR("no features should be dumped");
+    }
+    FeatureDumpFixture() : IDumpFeatureVisitor() {}
+};
+
+struct Labels {
```
Comment (geirst) [inline searchlib/src/tests/features/nns_closeness/nns_closeness_test.cpp]: Labels, NoLabels and SingleLabel are now duplicated in 3 unit tests. Please move them to searchlib/src/vespa/searchlib/fef/test and put them in namespace search::fef::test.

**Review by geirst — APPROVED** [review]
👍 Please move some duplicated test helper code first.

File: searchlib/src/tests/features/nns_closeness/nns_closeness_test.cpp (hunk 1 above)
Comment (arnej27959) [reply searchlib/src/tests/features/nns_closeness/nns_closeness_test.cpp]: Labels etc. moved to fef::test now.

**Comment by arnej27959** [comment]
will merge after factory is green.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**